spec/datadog/appsec/actions_handler_spec.rb

```diff
@@ -97,4 +101,124 @@
       end
     end
   end
+
+  describe '.generate_stack' do
+    let(:generate_stack_action) { { 'stack_id' => 'foo' } }
+    let(:trace_op) { Datadog::Tracing::TraceOperation.new }
+    let(:trace_op_metastruct) { nil }
+    let(:span_op) { Datadog::Tracing::SpanOperation.new('span_test') }
+    let(:span_op_metastruct) { nil }
+    let(:context) { OpenStruct.new(trace: trace_op, span: span_op) }
+    let(:stack_trace_enabled) { true }
+    let(:max_collect) { 0 }
+
+    let(:stack_key) { Datadog::AppSec::Ext::TAG_STACK_TRACE }
+    let(:exploit_category) { Datadog::AppSec::Ext::EXPLOIT_PREVENTION_EVENT_CATEGORY }
+
+    before do
+      Datadog.configure do |c|
```

Comment:
FYI if you go this way - you will need to clean the settings afterwards (see other tests where we do that)